compiler/pash_runtime_daemon.py

```diff
@@ -225,11 +231,13 @@ def parse_and_run_cmd(self, input_cmd):
                 f'Unsupported command: {input_cmd}'))
 
     def wait_input(self):
-        input_cmd = ""
-        with open(self.in_filename) as fin:
-            input_cmd = fin.read()
-            input_cmd = input_cmd.rstrip()
-        return input_cmd
+        cmd = ""
+        if not self.reader_pipes_are_blocking:
+            while not cmd:
```

Comment:
We should move this while internally in get_next_cmd(). The scheduler should not be exposed to those things

Reply:
I think it makes sense to have this in the scheduler because the scheduler initializes the UnixPipeReader as nonblocking so it should deal with the behavior of nonblocking reading.